[ci][api][webui] Refactor rubocop offenses

mdeniz opened the pull request:
Fixed rubocop offenses for cop Lint/AssignmentInCondition.

mdeniz commented:
@ChrisBr, @bgeuken, @adrianschroeter: Can you guys please review this?

bgeuken commented on src/api/app/helpers/branch_package.rb:
```diff
@@ -406,7 +409,8 @@ def check_for_update_project(p)
         # We have no package in the update project yet, but sources are reachable via project link
-        if update_project.develproject and up = update_project.develproject.find_package(pkg_name)
+        up = update_project.develproject.find_package(pkg_name) if update_project.develproject
+        if update_project.develproject && up
```
No need to check for update_project.develproject here. If up got set it's there.

mdeniz replied:
I don't agree. If ```update_project.develproject``` is false or nil then *up* is not set... so it will fail with a ```NameError: undefined local variable or method 'up' for main:Object``` in the ```if``` statement if I only use it directly without checking first for ```update_project.develproject```

bgeuken commented on src/api/app/helpers/maintenance_helper.rb:
```diff
@@ -268,8 +271,11 @@ def import_channel(channel, pkg, targetRepo = nil)
-          if a = prj.find_attribute('OBS', 'UpdateProject') and a.values[0]
+        attrib = b.attributes.get_attribute('project')
+        prj = Project.get_by_name(attrib.to_s) if attrib
+        if attrib && prj
```
same here: if prj should be enough

mdeniz replied:
Same here again, prj will be not set and then it will raise an error

bgeuken commented on src/api/app/models/bs_request.rb:
```diff
@@ -381,7 +381,8 @@ def changestate_accepted(opts)
           attrib = source_project.attribs.where(attrib_type_id: at.id).first
-          if attrib && v=attrib.values.first
+          v = attrib.values.first if attrib
+          if attrib && v
```
same here

mdeniz replied:
Same here again, *v* will be not set and then it will raise an error

bgeuken commented on src/api/app/models/bs_request_action.rb:
```diff
@@ -839,20 +839,20 @@ def check_action_permission_target!
-
-      if tpkg && (a = tpkg.find_attribute('OBS', 'RejectRequests') and a.values.first)
+      a = tpkg.find_attribute('OBS', 'RejectRequests') if tpkg
+      if tpkg && a && a.values.first
```
and here

mdeniz replied:
Here is slightly different, I have to take care of *tpkg* also...

bgeuken commented on src/api/app/models/bs_request_action_maintenance_incident.rb:
```diff
@@ -97,7 +97,8 @@ def _merge_pkg_into_maintenance_incident(incidentProject, source_project, source
       # accept branching from former update incidents or GM (for kgraft case)
-      if linkinfo and linkprj = Project.find_by_name(linkinfo['project'])
+      linkprj = Project.find_by_name(linkinfo['project']) if linkinfo
+      if linkinfo && linkprj
```
here as well:)

mdeniz replied:
Same here again, *linkprj* will be not set and then it will raise an error

bgeuken commented on src/api/test/activexml_matcher.rb:
```diff
@@ -175,7 +175,10 @@ def self.match(node, conditions)
         p = node
-        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
+        while p
+          throw :found, true if p && match(p, conditions[:ancestor])
```
I think the 'if p' is already covered by the while loop, right?

mdeniz replied:
Yes I think so

bgeuken commented:
LGTM

marcus-h commented:
Just a very minor nit-pick: sometimes the wrong commit prefix is used
(since some commits also affect the api). Recently for this kind of
commit the prefix "[ci]" was used (maybe we could also use something like
"[ci,api,webui]"?).

Minor remark about commit 658ee9141e243f6d20cd36cff749dc068ab7b885:

> diff --git a/src/api/test/activexml_matcher.rb b/src/api/test/activexml_matcher.rb
> index 97b1bf0..48e53ff 100644
> --- a/src/api/test/activexml_matcher.rb
> +++ b/src/api/test/activexml_matcher.rb
> @@ -175,7 +175,10 @@ def self.match(node, conditions)
>      if conditions[:ancestor]
>        return false unless catch :found do
>          p = node
> -        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
> +        while p
> +          throw :found, true if p && match(p, conditions[:ancestor])

The "p" check can be omitted       ^

> +          p = p.parent
> +        end
>        end
>      end

Is this correct w.r.t. semantics? (the former loop started with "node.parent",
while the new code starts with "node")

mdeniz commented:
@bgeuken I have changed the way of checking the variables using now "defined?" and then check for the variable value

@marcus-h Renamed the commit's prefixes. And fixed the loop. Many thanks

adrianschroeter commented:
we wrote the tags so far as "[api][webui]" instead of the comma to allow easier grepping

mdeniz commented:
@adrianschroeter: Done. Renamed properly.

bgeuken commented on src/api/test/activexml_matcher.rb:
```diff
@@ -175,7 +175,10 @@ def self.match(node, conditions)
         p = node
-        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
+        while p
+          p = p.parent
```
This now omits the first state of p (== node)

mdeniz replied:
Yes. @marcus-h commented it here https://github.com/openSUSE/open-build-service/pull/1132#issuecomment-140434914
The first value of *p* should be an ancestor, not the node itself... isn't it?

marcus-h replied:
On 2015-09-16 04:23:44 -0700, Moisés Déniz Alemán wrote:
> > @@ -175,7 +175,10 @@ def self.match(node, conditions)
> >      if conditions[:ancestor]
> >        return false unless catch :found do
> >          p = node
> > -        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
> > +        while p
> > +          p = p.parent
> 
> Yes. @marcus-h commented it here https://github.com/openSUSE/open-build-service/pull/1132#issuecomment-140434914
> The first value of *p* should be an ancestor, not the node itself... isn't it?
> 
Yep (that's what the old code did/does). It could be translated as

p = node.parent
while p
  throw :found, true if match(p, conditions[:ancestor])
  p = p.parent
end

mdeniz replied:
I think that yours is better than mine. I'll change it. 
In my implementation *p* can be *nil* when calling *match*, and can be wrong.
